src/components/AuthForm.js

```diff
@@ -363,18 +366,49 @@ export default class AuthForm extends PureComponent<
 
   constructor(props: Object) {
     super(props);
-    const { settings: { authMethods } } = this.props;
-    const defaultAuth = authMethods[0];
     const { schema, uiSchema } = authSchemas[defaultAuth];
     this.state = {
       schema,
       uiSchema,
       formData: { authType: defaultAuth },
     };
+
+    const { history, getServerInfo } = this.props;
+    const server = (history && history[0]) || defaultKintoServer;
+    const anonymousAuth = { authType: "anonymous", server: server };
+
+    getServerInfo(anonymousAuth);
   }
 
+  getSupportedAuthMethods = () => {
+    const { session: { serverInfo: { capabilities } } } = this.props;
+    // Check which of our known auth implementations are supported by the server.
+    const knownAuthMethods = [
+      "basicauth",
+      "account",
+      "fxa",
+      "ldap",
+      "portier",
+      "anonymous",
+    ];
+    let supportedAuthMethods = [defaultAuth];
+    knownAuthMethods.forEach(authMethod => {
+      if (authMethod in capabilities) {
+        supportedAuthMethods.push(authMethod);
+      }
+    });
```

Comment:
nitpick: `knownAuthMethods.filter(a => a in capabilities)` ?

Reply:
And then not forget to add the "anonymous" method which isn't in the capabilities as you said. Looks like a good idea, thanks ;)